common-npm-packages/packaging-common/npm/npmregistry.ts

```diff
@@ -59,31 +60,32 @@ export class NpmRegistry implements INpmRegistry {
                 username = endpointAuth.parameters['username'];
                 password = endpointAuth.parameters['password'];
                 email = username; // npm needs an email to be set in order to publish, this is ignored on npmjs
-                password64 = (new Buffer(password).toString('base64'));
-                tl.setSecret(password64);
+                password64 = Buffer.from(password).toString('base64');
 
                 auth = nerfed + ':username=' + username + lineEnd;
                 auth += nerfed + ':_password=' + password64 + lineEnd;
                 auth += nerfed + ':email=' + email + lineEnd;
                 break;
             case 'Token':
                 const apitoken = endpointAuth.parameters['apitoken'];
+                tl.setSecret(apitoken);
                 if (!isVstsTokenAuth){
                     // Use Bearer auth as it was intended.
                     auth = nerfed + ':_authToken=' + apitoken + lineEnd;
                 } else {
                     // Azure DevOps does not support PATs+Bearer only JWTs+Bearer
                     email = 'VssEmail';
                     username = 'VssToken';
-                    password64 = (new Buffer(apitoken).toString('base64'));
-                    tl.setSecret(password64);
+                    password64 = Buffer.from(apitoken).toString('base64');
 
                     auth = nerfed + ':username=' + username + lineEnd;
                     auth += nerfed + ':_password=' + password64 + lineEnd;
                     auth += nerfed + ':email=' + email + lineEnd;
                 }
                 break;
         }
+        tl.setSecret(password);
```

Comment:
From the code above, only one of these lines will actually be used in any scenario. Can we use the .setSecret() method on an object that hasn't been set?

Reply:
yea the function setSecret does do a null/empty string check before doing anything.

/**
 * Registers a value with the logger, so the value will be masked from the logs.  Multi-line secrets are not allowed.
 *
 * @param val value to register
 */
function setSecret(val) {
    if (val) {
        if (val.match(/\r|\n/) && ("" + process.env['SYSTEM_UNSAFEALLOWMULTILINESECRET']).toUpperCase() !== 'TRUE') {
            throw new Error(exports.loc('LIB_MultilineSecret'));
        }
        exports.command('task.setsecret', {}, val);
    }
}